### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by momotsow**
You can find the features added for this Pull Request below.

- A method called current_user was created in your ApplicationController to make current user data available to all controllers. It returns the first user from the database.
- Forms were created to perform the following functions:
  - Creates a Post for the current_user.
  - Creates a comment for the current_user.
- Methods and views are added to Users to add likes to Posts.
- After saving the form, updates the posts' likes and comments counter.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -5,8 +5,11 @@
   # root "articles#index"
 
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show], param: :id
-  end 
+    resources :posts, only: [:index, :show, :new, :create], param: :id
+    resources :comments, only: [:new, :create]
+    resources :likes, only: [:create]
+  end
+end
```
Comment (MeqdamAlqudah) [inline config/routes.rb]: - [x] Please make sure that your routes.rb file is setup correctly, I think you forget to add do after the post resources I'm getting this error.
image

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
```
Comment (MeqdamAlqudah) [inline app/controllers/comments_controller.rb:1]: - [x]  Please the post controller should contain a new and a create method that handles post Post request, please add it to your controller 
image

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @momotsow , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
image

## Highlights
:+1: no linter errors 
:+1: your user index page looks cool

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (momotsow) [reply app/controllers/comments_controller.rb:1]: Hi @MeqdamAlqudah,

Thank you for your review. I have resolved these.

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @momotsow  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Made changes as requested by the previous reviewer 👍 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**